Utils/Game.py

```diff
@@ -1,13 +1,337 @@
+        self.screen.blit(text, text_rect)
+
+    def draw_material_diff(self):
+        ratio = (self.gb.black_captured_value + 10) / (self.gb.black_captured_value + 20 + self.gb.white_captured_value)
+        b_x, b_y = self.X_OFFSET - 2 * self.MATERIAL_CHART_WIDTH, self.Y_OFFSET
+        w_x, w_y = b_x, b_y + int(ratio * self.BOARD_SIZE)
+        b_x_diff, b_y_diff=self.MATERIAL_CHART_WIDTH, w_y
+        w_x_diff, w_y_diff = self.MATERIAL_CHART_WIDTH, self.Y_OFFSET + self.BOARD_SIZE-w_y
+        pygame.draw.rect(self.screen, (0, 0, 0), (b_x, b_y,b_x_diff, b_y_diff ))
+
+        pygame.draw.rect(self.screen, (255, 255, 255),
+                         (b_x, w_y,w_x_diff, w_y_diff))
+
+    def draw_utils(self):
+        self.draw_reset_button()
+        self.draw_undo_button()
+        self.draw_material_diff()
+
+    def reset_clicked(self, x, y):
+        button_rect = pygame.Rect(self.START_BUTTON_X, self.START_BUTTON_Y, self.BUTTON_WIDTH,
+                                  self.BUTTON_HEIGHT)
+        if button_rect.collidepoint(x, y):
+            self.reinitialize()
+
+    # TODO Implement undo_clicked
+    def undo_clicked(self):
+        pass
+
+    def main(self):
+        pygame.init()
+        pygame.font.init()
+        self.load_images()
+        timer = pygame.time.Clock()
+        pygame.display.set_caption('Chess App')
+
+        while self.running:
+            timer.tick(self.fps)
+            self.screen.fill(self.BEIGE)
+            self.draw_board()
+            self.draw_pieces()
+            self.draw_utils()
+            self.fps_counter += 1
+            for event in pygame.event.get():
+                # quit
+                if event.type == pygame.QUIT:
+                    self.running = False
+
+                if event.type == pygame.MOUSEBUTTONDOWN:
+                    x, y = pygame.mouse.get_pos()
+                    self.reset_clicked(x, y)
+
+                # if game not ended
+                if not self.mate and not self.stalemate:
+                    # and we don't have a promotion
+                    if not self.gb.promoting_piece:
+                        # selecting a piece
+                        if event.type == pygame.MOUSEBUTTONDOWN:
+                            x, y = pygame.mouse.get_pos()
+                            self.select_piece(x, y)
+                            self.update_possible_moves()
+
+                        # if we have selected piece we can make a move
+                        if event.type == pygame.MOUSEBUTTONUP and self.selected_piece:
+                            x, y = pygame.mouse.get_pos()
+                            self.make_move(x, y)
+
+            if self.mate:
+                self.show_message('white' if self.gb.curr_player == 'black' else 'black', None)
+
+            elif self.stalemate:
+                self.show_message(None, "Stalemate")
+
+            if self.gb.promoting_piece:
+                self.show_promotion(self.gb.promoting_piece.position)
```

Comment:
`show_promotion_options`? More descriptive and if I understand correctly, this method should show options of promoting a pawn.